### PR TITLE
Resolve #5 -- Using the download from Google drive. Converts projects to x86

### LINKED ISSUE #5: Server unavailable causes crash
https://github.com/LeagueSandbox/LeagueSandboxAutoSetup/blob/da80da76c8ddb869862b6501df498bd3889e4072/League%20Sandbox%20Auto%20Setup/Form1.cs#L146

As the url is no longer working, it needs to be changed to a new URL.
Also, add error messages for such cases.

### PR DESCRIPTION
**Pull request by samuelGrahame**
Resolve #5

### CODE REVIEW

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 1]
```diff
@@ -36,11 +51,17 @@ private void OnAbortSuccessfully()
             startButton.Enabled = true;
             abortText.Visible = false;
         }
-
+        bool convertProjectsToX86 = false;
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs]: I don't think you need it to be class variable. Also, if you indeed need it, please define it at the start of the class.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 2]
```diff
@@ -49,7 +70,7 @@ private void startButton_Click(object sender, EventArgs e)
                 _setupStarted = true;
             }
             else
-            {
+            {
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs]: Delete added spaces after `{`

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 3]
```diff
@@ -117,68 +138,171 @@ private void startCloningRepositories()
...
                 }
                 cloningProgressLabel.Invoke(new Action(() =>
                 {
                     cloningProgressLabel.Text = "✔️";
 
+                    
+
+                    if(convertProjectsToX86)
+                    {
+                        convertProjectsToX86AterCloning(cloningPath);
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs]: Change `Ater` to `After`.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 4]
```diff
@@ -117,68 +138,171 @@ private void startCloningRepositories()
...
+                    }
+
                     startDownloadingClient();
                 }));
             }).Start();
         }
 
+        private void convertProjectsToX86AterCloning(string cloningPath)
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs]: Change `Ater` to `After`.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 5]
```diff
@@ -117,68 +138,171 @@ private void startCloningRepositories()
...
+            if (Directory.Exists(Path.Combine(installDirectoryText.Text, "League_Sandbox_Client")))
+            {                
+                downloadingProgressLabel.Text = "✔️";
+                startUnzippingClient();
+                return;
+            }
+
 
-            downloadingProgressLabel.Text = "--";
+            // downloadingProgressLabel.Text = "--";
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs]: Remove commented code.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 6]
```diff
@@ -117,68 +138,171 @@ private void startCloningRepositories()
...
 
-            System.Windows.Forms.Timer timer = new System.Windows.Forms.Timer();
-            timer.Interval = 100;
+            downloadingProgressLabel.Text = $"Please download .7z to: {installDirectoryText.Text} and name it: {Client_Folder_Name}.7z";
 
-            // register download ended event
-            DownloadManager.Instance.DownloadEnded += (_, _2) =>
+            Clipboard.SetText(localFilePath);
+
+            while (!File.Exists(localFilePath))
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs:239]: There are certainly better ways. (I think https://docs.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher.created?view=netcore-3.1 should be better)

This is not crucial currently. (I'll open an issue upon merge)

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 7]
```diff
@@ -117,68 +138,171 @@ private void startCloningRepositories()
...
-                downloadingProgressLabel.Text = $"{Math.Round(downloader.Progress, 2)}% - {Math.Round(downloader.Rate / 1024 / 1024, 2)} MB/s";
-            };
-            timer.Start();
+                Thread.Sleep(1);
+            }
+
+            downloadingProgressLabel.Text = "✔️";
+            startUnzippingClient();
+
+            //System.Windows.Forms.Timer timer = new System.Windows.Forms.Timer();
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs]: Delete commented code.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 8]
```diff
@@ -309,11 +440,14 @@ private void startVisualStudioFirstRun()
                 File.Delete(configNewPath);
             }
             JObject json = JObject.Parse(templateString);
-            json["clientLocation"] = leagueInstallFolder;
+            //C:\LeagueSandbox\League_Sandbox_Client\League-of-Legends-4-20\RADS\solutions\lol_game_client_sln\releases\0.0.1.68\deploy
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs:406]: Delete commented path.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 9]
```diff
@@ -232,6 +356,13 @@ private void startSettingUpTestbox()
...
             }
 
+            if(Directory.Exists(Path.Combine(installDirectoryText.Text, "LeagueUI")))
+            {
+                installingTestboxLabel.Text = "✔️";
+                startVisualStudioFirstRun();
+                return;
+            }
+
             String testboxLink = "http://gamemakersgarage.com/LeagueUI.7z";
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs:329]: This link will eventually break too. Not crucial. (I'll open an issue)

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 10]
```diff
@@ -117,68 +138,171 @@ private void startCloningRepositories()
                     }
                 };
                 options.BranchName = "Indev";
-                options.RecurseSubmodules = true;
+                options.RecurseSubmodules = false;
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs:139]: Why? Is there a problem with this recursive cloning?

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 11]
```diff
@@ -26,7 +28,20 @@ public leagueSandboxAutoSetupForm()
 
             this.FormClosing += (_, _2) =>
             {
-                Environment.Exit(0);
+                try
+                {
+                    _abortInitiated = true;
+                    abortText.Visible = true;
+                    startButton.Text = "Start";
+                    startButton.Enabled = false;
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs:34]: Why do you need to change variables before the form close?

File: League Sandbox Auto Setup/appsettings.json [hunk 12]
```diff
@@ -0,0 +1,12 @@
+{
+  "Core": {
```
Comment (danil179) [inline League Sandbox Auto Setup/appsettings.json]: Do you use it? I think `SegmentDownloader.Core.Instrumentation;` is not used.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs [hunk 13]
```diff
@@ -12,6 +12,8 @@
 using System.Diagnostics;
 using Newtonsoft.Json.Linq;
 using League_Sandbox_Auto_Setup.Util;
+using System.Net;
+using SegmentDownloader.Core.Instrumentation;
```
Comment (danil179) [inline League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs]: Do you use `SegmentDownloader.Core.Instrumentation;`?

**Review by danil179 — CHANGES_REQUESTED** [review]
Some comments.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs (hunk 10 above)
Comment (SamG-Coder) [reply League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs:139]: Yeah, everytime I ran it, it crashed.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs (hunk 11 above)
Comment (SamG-Coder) [reply League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs:34]: If you clicked the close button and it threw an exception. Once completed.

File: League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs (hunk 4 above)
Comment (SamG-Coder) [reply League Sandbox Auto Setup/leagueSandboxAutoSetupForm.cs]: Okay will fix

**Review by danil179 — APPROVED** [review]
I think that this is better than the current state. Approved.

**Pull request merged**